lib/commands/web.js

```diff
@@ -2,12 +2,39 @@ import { iosCommands } from 'appium-ios-driver';
 import { retryInterval } from 'asyncbox';
 import log from '../logger';
 import _ from 'lodash';
+import B from 'bluebird';
 
 
 let extensions = {};
 
 Object.assign(extensions, iosCommands.web);
 
+extensions.getExtraNativeWebTapOffset = async function () {
+  let offset = 0;
+
+  // first try to get tab offset
+  let els = await this.findNativeElementOrElements('-ios predicate string', "wdName LIKE '*, Tab' AND visible = 1", true);
+  if (els.length) {
+    els = (await B.all(_.uniq(els).map((el) => this.getNativeRect(el.ELEMENT)))).map((el) => el.height);
+    log.debug(`Found ${els.length} tabs with heights ${els.join(', ')}. Setting offset to ${els[0]}`);
+    offset = els[0];
+  }
+
+  // next try to see if there has been scrolling
+  els = await this.findNativeElementOrElements('-ios predicate string', "wdName = 'ReloadButton'", true);
```

Comment:
also, I assume it would be faster to search for a single element rather than all elements, especially after https://github.com/facebook/WebDriverAgent/pull/641 is merged